src/query-builder.ts

```diff
@@ -1,161 +1,160 @@
-      .filter(([name, _value]) => name !== 'fields') // Fields is handled separately from the rest
-      .reduce((acc: string[], [name, value]) => {
-        const argName = name as keyof QueryIO[typeof typedEntityName]['input'] // type the argument name, some queries do not allow doing where or other filters, so we need to type it
-        const argValues = value as FilterValue<typeof typedEntityName, keyof QueryIO[typeof typedEntityName]['input']>
+  const filterArgs = Object.entries(filters)
+    .filter(([name, _value]) => name !== 'fields') // Fields is handled separately from the rest
+    .reduce((acc: string[], [name, value]) => {
+      const argName = name as keyof EntityQueryMap[typeof typedEntityName]['input'] // type the argument name, some queries do not allow doing where or other filters, so we need to type it
+      const argValues = value as FilterValue<typeof typedEntityName, keyof EntityQueryMap[typeof typedEntityName]['input']>
 
-        const processed = this.processFilter(typedEntityName, argName, argValues)
-        if (processed) acc.push(processed)
+      const processed = processFilter(typedEntityName, argName, argValues)
+      if (processed) acc.push(processed)
 
-        return acc
-      }, [])
-      .join(', ')
-    // Handle edge case: don't include empty parentheses when args is empty
-    const filtersForQuery = filterArgs ? `(${filterArgs})` : ''
+      return acc
+    }, [])
+    .join(', ')
+  // Handle edge case: don't include empty parentheses when args is empty
+  const filtersForQuery = filterArgs ? `(${filterArgs})` : ''
 
-    const queryForEntity = `${String(entityName)}${filtersForQuery} {
-      ${fields.join('\n                ')}
-    }`
+  const queryForEntity = `${String(entityName)}${filtersForQuery} {
+    ${fields.join('\n                ')}
+  }`
 
-    return queryForEntity
-  }
+  return queryForEntity
+}
 
-  /**
-   * Builds a complete GraphQL query string from the input
-   * @param input - The query input to build
-   * @returns A complete GraphQL query string
-   */
-  build<T extends QueryInput>(input: T & Record<Exclude<keyof T, keyof QueryInput>, never>): string {
-    const entities = Object.entries(input)
-    const queryStr = `
-      query {
-        ${entities
-          .map(([entity, filtersForQuery]) => {
-            const entityName = entity as keyof QueryIO
-            const filters = filtersForQuery as QueryIO[typeof entityName]['input']
-            const parsedQuery = this.parseEntityQuery({
-              entityName,
-              filters
-            })
-            return parsedQuery
-          }).join('\n          ')}
-      }  
-      `
-    return queryStr
-  }
+/**
+ * Builds a complete GraphQL query string from the input
+ * @param input - The query input to build
+ * @returns A complete GraphQL query string
+ */
+const build = <T extends QueryInput>(input: T & Record<Exclude<keyof T, keyof QueryInput>, never>): string => {
```

Comment:
is there a reason to use arrow functions here? In general its better to use declared functions for better error tracking